packages/components/src/modal/index.tsx

```diff
@@ -223,7 +227,9 @@ function UnforwardedModal(
 					ref={ useMergeRefs( [
 						constrainedTabbingRef,
 						focusReturnRef,
-						focusOnMountRef,
+						focusOnMount !== 'firstContentElement'
+							? focusOnMountRef
+							: noopRef,
```

Comment:
The hook requires that the parameter is a `ref` thus the `noopRef`.

Reply:
It does seem to be what the typing says yet have a look at how it's used in `useDialog`:
https://github.com/WordPress/gutenberg/blob/a2e4ac34eafe3602c5d4212b7573539fabd5c02a/packages/compose/src/hooks/use-dialog/index.ts#L85-L90

Reply:
That's strange. I could have sworn I tried that and it didn't work. 

I've updated now and it does in fact work as you said. Thank you.